src/app/inicio/hook.ts

```diff
@@ -0,0 +1,68 @@
+"use client";
+
+import { useUser } from "@auth0/nextjs-auth0/client";
+import { AxiosError } from "axios";
+import { useEffect, useRef, useState } from "react";
+import { toast } from "react-toastify";
+
+import { ErrorMappings, ITransaction } from "@/interfaces";
+import api from "@/services/api";
+import { AXIOS_ERROR_400, AXIOS_ERROR_404 } from "@/utils/constants";
+
+export const useHome = () => {
+  const [transactions, setTransactions] = useState<ITransaction[]>([]);
+  const [isloading, setIsLoading] = useState<boolean>(false);
+  const observerTarget = useRef(null);
+  const { user } = useUser();
+
+  async function loadTransactions(): Promise<string | void> {
+    try {
+      const { data } = await api.get<ITransaction>(`/transaction/${user?.sub}`);
+      setIsLoading(true);
+      if (Array.isArray(data) && data.length !== 0) {
+        setIsLoading(false);
+        setTransactions(data);
+      } else {
+        setIsLoading(false);
+        toast.error("Nenhuma transação foi encontrada.");
+      }
```

Comment:
Está sendo inútil isso, pois não está usando em nenhum lugar.

Reply:
verdade pois já tem uma mensagem no page que verifica e mostra quando não há nenhuma transação. Então removi as linhas acima.